tests/test_trainer.py

```diff
@@ -411,5 +413,98 @@ class CurrentTestModel(
     trainer.test()
 
 
+def test_trainer_max_steps_and_epochs(tmpdir):
+    """Verify model trains according to speficied max steps"""
+    tutils.reset_seed()
```

Comment:
personally I would cut this out as a "private func" as it is the same for both tests
```
def _init_model(tempdir):
    tutils.reset_seed()
    model, _ = tutils.get_model()

    # define train epoch to 5% of data
    train_percent = 0.05
    # get number of samples in 1 epoch
    num_train_samples = math.floor(len(model.train_dataloader()) * train_percent)

    trainer_options = dict(
        default_save_path=tmpdir,
        train_percent_check=train_percent,
    )
    return model, trainer_options
```
and later in tests just :
```
def test_trainer_max_steps_and_epochs(tmpdir):
    """Verify model trains according to speficied max steps"""
    model, trainer_options = _init_model(tmpdir)
    ...
```